test/jdk/sun/net/www/http/KeepAliveCache/B8291637.java

```diff
@@ -125,20 +130,26 @@ public void run() {
         }
     }
 
-    public static void main(String[] args) throws Exception {
-        Server server = new Server(args[0]);
+    public static void runTest(String param) throws Exception {
+        Server server = new Server(param);
         int port = server.getPort();
         server.start();
-        URL url = new URL("http://127.0.0.1:" + Integer.toString(port) + "/");
+        URL url = URIBuilder.newBuilder()
+                .scheme("http")
+                .loopback()
+                .port(port)
+                .path("/")
+                .toURL();
         HttpURLConnection urlc = (HttpURLConnection) url.openConnection();
-        InputStream i = urlc.getInputStream();
-        int c,count=0;
-        byte[] buf = new byte[256];
-        while ((c=i.read(buf)) != -1) {
-            count+=c;
+        int count=0;
+        try (InputStream i = urlc.getInputStream()) {
+            int c;
+            byte[] buf = new byte[256];
+            while ((c = i.read(buf)) != -1) {
+                count += c;
+            }
         }
-        i.close();
-        System.out.println("Read " + count );
+        System.out.println("Read " + count);
         try {
```

Comment:
Maybe the `try {` line should be moved just after `server.start()`

Reply:
good point. I enhanced this a little bit more. 😄